modules/javafx.graphics/src/main/java/javafx/scene/SpotLight.java

```diff
@@ -0,0 +1,231 @@
+            public NGNode doCreatePeer(Node node) {
+                return ((SpotLight) node).doCreatePeer();
+            }
+
+            @Override
+            public void doUpdatePeer(Node node) {
+                ((SpotLight) node).doUpdatePeer();
+            }
+        });
+    }
+
+    {
+        // To initialize the class helper at the beginning each constructor of this class
+        SpotLightHelper.initHelper(this);
+    }
+
+    /**
+     * Creates a new instance of {@code SpotLight} class with a default {@code Color.WHITE} light source.
+     */
+    public SpotLight() {
+        super();
+    }
+
+    /**
+     * Creates a new instance of {@code SpotLight} class using the specified color.
+     *
+     * @param color the color of the light source
+     */
+    public SpotLight(Color color) {
+        super(color);
+    }
+
+
+    /**
+     * The direction the spotlight is facing. The vector need not be normalized.
+     *
+     * @defaultValue {@code Point3D(0, 0, -1)}
+     */
+    private ObjectProperty<Point3D> direction;
+
+    public final void setDirection(Point3D value) {
+        directionProperty().set(value);
+    }
+
+    private static final Point3D DEFAULT_DIRECTION = NGSpotLight.getDefaultDirection();
+
+    public final Point3D getDirection() {
+        return direction == null ? DEFAULT_DIRECTION : direction.get();
+    }
+
+    public final ObjectProperty<Point3D> directionProperty() {
+        if (direction == null) {
+            direction = new SimpleObjectProperty<>(this, "direction", DEFAULT_DIRECTION) {
+                @Override
+                protected void invalidated() {
+                    NodeHelper.markDirty(SpotLight.this, DirtyBits.NODE_LIGHT);
+                }
+            };
+        }
+        return direction;
+    }
+
+
+    /**
+     * The angle of the spotlight's inner cone. Surfaces whose angle to the light's origin is less than this angle
+     * receive the full light's intensity. At larger angles, the light intensity starts to drop.
+     * <p>
+     * The angle is clamped between 0 and 180.
+     *
+     * @defaultValue 0
+     */
+    private DoubleProperty innerAngle;
+
+    public final void setInnerAngle(double value) {
+        innerAngleProperty().set(value);
+    }
+
+    private static final double DEFAULT_INNER_ANGLE = NGSpotLight.getDefaultInnerAngle();
+
+    public final double getInnerAngle() {
+        return innerAngle == null ? DEFAULT_INNER_ANGLE : innerAngle.get();
+    }
+
+    public final DoubleProperty innerAngleProperty() {
+        if (innerAngle == null) {
+            innerAngle = getLightDoubleProperty("innerAngle", DEFAULT_INNER_ANGLE);
+        }
+        return innerAngle;
+    }
+
+
+    /**
+     * The angle of the spotlight's outer cone. Surfaces whose angle to the light's origin is greater than this angle
+     * receive no light. At smaller angles, the light intensity starts to increase.
+     * <p>
+     * The angle is clamped between 0 and 180.
```

Comment:
between innerAngle and 180?

Reply:
If we aren't going to clamp the values to be between `innerAngle` and 180 then this sentence should just be removed.